### PR TITLE
Enable `--proxy-to-worker` under node.

### PR DESCRIPTION
**Pull request by sbc100**
At least enough to run hello world.  The idea here is not so much that folks really want use `--proxy-to-worker` under node, but more for testing, so we can run tests where the module is instantiated on a worker.

In particular we can write tests for issues such as #19577 that run under node.

### CODE REVIEW

File: src/proxyWorker.js [hunk 1]
```diff
@@ -4,27 +4,6 @@
  * SPDX-License-Identifier: MIT
  */
 
-if (typeof console == 'undefined') {
```
Comment (sbc100) [inline src/proxyWorker.js:7]: I'm not sure what this code could have been for.. there are no environments that I know of that supports workers but don't have a console object.

File: emcc.py [hunk 2]
```diff
@@ -4166,10 +4165,10 @@ def generate_worker_js(target, js_target, target_basename):
 
 def worker_js_script(proxy_worker_filename):
   web_gl_client_src = read_file(utils.path_from_root('src/webGLClient.js'))
-  idb_store_src = read_file(utils.path_from_root('src/IDBStore.js'))
-  proxy_client_src = read_file(utils.path_from_root('src/proxyClient.js'))
-  proxy_client_src = do_replace(proxy_client_src, '{{{ filename }}}', proxy_worker_filename)
-  proxy_client_src = do_replace(proxy_client_src, '{{{ IDBStore.js }}}', idb_store_src)
+  proxy_client_src = shared.read_and_preprocess(utils.path_from_root('src/proxyClient.js'), expand_macros=True)
+  if not os.path.dirname(proxy_worker_filename):
+    proxy_worker_filename = './' + proxy_worker_filename
```
Comment (kripken) [inline emcc.py:4170]: What is this `./` addition doing? And why only if it isn't a dirname?

**Review by kripken — COMMENTED** [review]
lgtm % question

File: emcc.py (hunk 2 above)
Comment (sbc100) [reply emcc.py:4170]: Under node `new Worker('foo.js')` doesn't seem to work buy `new Worker('./foo.js')` does.

Comment (kripken) [reply emcc.py:4170]: This is worth a comment then, I think, as it seems non-obvious in the code.

**Pull request merged**